### PR TITLE
Update Invoke-AzVMRunCommand.md NOTES

### PR DESCRIPTION
**Pull request by goldjg**
Add security information in the NOTES section.

## Description

Add information about the persistence of script files
on the VM host, and the potential exposure of commandline arguments (already raised with MSRC - product team will be making some changes to the agent and some documentation changes).

## Checklist

- [X] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by msJinLei** [comment]
@Drewm3 Could you help to review the PR, thanks

**Comment by goldjg** [comment]
Any update on the review?

**Comment by Drewm3** [comment]
@nikhilpatel909, could you please take a look at this PR and let me know if it looks good.

File: src/Compute/Compute/help/Invoke-AzVMRunCommand.md [hunk 1]
```diff
@@ -260,4 +260,9 @@ This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable
 
 ## NOTES
 
+This command leaves a copy of every script that it runs on the Linux VM it runs against each time it is invoked.
```
Comment (vivlingaiah) [inline src/Compute/Compute/help/Invoke-AzVMRunCommand.md:263]: Script belonging to only the latest invocation is retained. The previous scripts are actually deleted. It was security Ask.

**Comment by goldjg** [comment]
When was that change made? I raised a case with MSRC on this in November 2022, they said they would ask the product team to implement a setting that would override the default behaviour and always clear up at end of invocation. Didn’t want to change the default because some customers relied on how it already worked.

Regards,
Graham

**Comment by goldjg** [comment]
> When was that change made? I raised a case with MSRC on this in November 2022, they said they would ask the product team to implement a setting that would override the default behaviour and always clear up at end of invocation. Didn’t want to change the default because some customers relied on how it already worked. Regards, Graham
> […](#)

https://cirriustech.co.uk/blog/azure-linux-vm-agent-secrets-leakage/

**Comment by nikhilpatel909** [comment]
this change was deployed in November 2022

**Comment by goldjg** [comment]
> this change was deployed in November 2022

By any chance was this as a result of:
Submission number
VULN-080939
Case number
76094

I raised that on 17th November initially.

There was also supposed to be a future change to add an option to always remove artefacts after invocation.

**Comment by BethanyZhou** [comment]
@nikhilpatel909 @Drewm3 what's your opinions about this PR? should we accept this notes?

**Comment by nikhilpatel909** [comment]
@BethanyZhou As [vivlingaiah] suggested above "This command leaves a copy of every script that it runs on the Linux VM it runs against each time it is invoked." it was a security risk, hence Script belonging to only the latest invocation is retained. The previous scripts are actually deleted. It was security Ask. We should not accept this PR. I am also confirming this with the developer.

**Comment by BethanyZhou** [comment]
Thanks @nikhilpatel909

**Comment by msJinLei** [comment]
@goldjg Thanks for your contribution. We close the PR now and welcome to continue to Azure PowerShell!

**Pull request closed without merging**